src/libs/Navigation/Navigation.ts

```diff
@@ -49,37 +54,32 @@ const getTopmostReportId = (state = navigationRef.getState()) => originalGetTopm
 // Re-exporting the getTopmostReportActionID here to fill in default value for state. The getTopmostReportActionID isn't defined in this file to avoid cyclic dependencies.
 const getTopmostReportActionId = (state = navigationRef.getState()) => originalGetTopmostReportActionId(state);
 
-/**
- * Method for finding on which index in stack we are.
- * @param {Object} route
- * @param {Number} index
- * @returns {Number}
- */
-const getActiveRouteIndex = function (route, index) {
-    if (route.routes) {
-        const childActiveRoute = route.routes[route.index || 0];
-        return getActiveRouteIndex(childActiveRoute, route.index || 0);
+/** Method for finding on which index in stack we are. */
+function getActiveRouteIndex(stateOrRoute: StateOrRoute, index?: number): number | undefined {
+    if ('routes' in stateOrRoute && stateOrRoute.routes) {
```

Comment:
Because stateOrRoute is a union of different types we need to first narrow down the type with `'routes' in stateOrRoute` and later make the check `stateOrRoute.routes`